auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,376 @@
+        string='Auto. E-mail on backup fail',
+        help=(
+            "If you check this option you can choose to automaticly"
+            " get e-mailed when the backup to the external server failed."
+        )
+    )
+    emailtonotify = fields.Char(
+        string='E-mail to notify',
+        help=(
+            "Fill in the e-mail where you want to be"
+            " notified that the backup failed on the FTP."
+        )
+    )
+    lasterrorlog = fields.Text(
+        string='E-mail to notify',
+        help=(
+            "Fill in the e-mail where you want to be"
+            " notified that the backup failed on the FTP."
+        )
+    )
+
+    @api.multi
+    def _check_db_exist(self):
+        for rec in self:
+            db_list = db.exp_list(True)
+            if rec.name in db_list:
+                return True
+            return False
+
+    _constraints = [
+        (
+            _check_db_exist,
+            _('Error ,No such database exists'), ['name']
+        )
+    ]
+
+    @api.multi
+    def test_sftp_connection(self):
+        confs = self.search([])
+        # Check if there is a success or fail and write messages
+        messageTitle = ""
+        messageContent = ""
+        conn_success = False
+        for rec in confs:
+            try:
+                conn_success = True
+                ipHost = rec.sftpip
+                portHost = rec.sftpport
+                usernameLogin = rec.sftpusername
+                passwordLogin = rec.sftppassword
+                # Connect with external server over SFTP, so we know sure that
+                # everything works.
+                srv = pysftp.Connection(host=ipHost, username=usernameLogin,
+                                        password=passwordLogin, port=portHost)
+                srv.close()
+                # We have a success.
+                messageTitle = _("Connection Test Succeeded!")
+                messageContent = _(
+                    "Everything seems properly set up for FTP back-ups!")
+            except Exception as e:
+                conn_success = False
+                messageTitle = _("Connection Test Failed!")
+                if len(rec.sftpip) < 8:
+                    messageContent += _(
+                        "\nYour IP address seems to be too short.\n")
+                messageContent += _("Here is what we got instead:\n")
+        if not conn_success:
+            raise except_orm(
+                _(messageTitle), _(
+                    messageContent + "%s") %
+                tools.ustr(e))
+        else:
+            raise UserError(_(messageTitle), _(messageContent))
+
+    @api.model
+    def schedule_backup(self):
+
+        for rec in self.search([]):
+            if not os.path.isdir(rec.bkp_dir):
+                os.makedirs(rec.bkp_dir)
+            # Create name for dumpfile.
+            bkp_file = '%s_%s.dump.zip' % (
+                time.strftime('%d_%m_%Y_%H_%M_%S'),
+                rec.name)
```

Comment:
IMHO we should not rely on the database name for the generated file name. As I said before, you should not be able to choose a database, so instead I think it's better if the sysadmin just creates a folder for each database where all files are called the same.

I mean, instead of saving in `~/backup/2016_02_29_15_31_21_mydb.dump.zip`, force the sysadmin to save them in `~/backup/mydb/2016_02_29_15_31_21.dump.zip`, so he can use the directory tree structure he wants and all files are named equally. Better for scripting etc.